skycatalogs/objects/sso_object.py

```diff
@@ -0,0 +1,178 @@
+                 mjd):
+        super().__init__(ra, dec, id, self._type_name, belongs_to,
+                         belongs_index)
+        self._mjd = mjd
+
+    @property
+    def mjd(self):
+        return self._mjd
+
+    def _get_sed(self, mjd=None):
+        '''
+        returns a SED and magnorm
+        mjd is required
+        '''
+        if SsoObject._solar_sed is None:
+            SsoObject._solar_sed =\
+                self._belongs_to._sky_catalog._sso_sed_factory.create()
+            # For magnorm use the magnitude from Sorcha.  Can it be used
+            # directly or are there other effects to be applied?
+            # Have to find it by looking for entry for this id, this mjd
+            # Do we look for specific entry or do we allow interpolation?
+        return SsoObject._solar_sed, self.get_native_attribute('trailed_source_mag')
+
+    def get_gsobject_components(self, gsparams=None, rng=None,
+                                streak=True):
+        trail_width = 1.e-6   # in arcseconds
+        exposure = self._belongs_to._exposure
+
+        if gsparams is not None:
+            gsparams = galsim.GSParams(**gsparams)
+
+        if streak:
+            # For a streak make galsim.Box with length the angular distance
+            # in arcseconds from initial ra, dec to final
+            # Make the width very, very small
+            # Then rotate appropriately
+            ra = self.ra
+            dec = self.dec
+            # Convert from (arc?)degrees/day to degrees/sec
+            ra_rate = self.get_native_attribute('ra_rate')/SECONDS_PER_DAY
+            # Take out factor of cos(dec).
+            ra_rate_raw = ra_rate/np.cos(dec)
+            # ra_rate = self.get_native_attribute('ra_rate')/24.0
+            dec_rate = self.get_native_attribute('dec_rate')/SECONDS_PER_DAY
+            # ra_final is approximate since really ra_rate is a function
+            # of dec, but average dec_rate is small so
+            ra_final = ra + ra_rate_raw * exposure
+            dec_final = dec + dec_rate * exposure
+
+            init_v = UnitVector3d(LonLat.fromDegrees(ra, dec))
+            final_v = UnitVector3d(LonLat.fromDegrees(ra_final, dec_final))
+            length = np.degrees(np.arccos(init_v.dot(final_v))) * 3600.0
+            gobj = galsim.Box(length, trail_width, gsparams=gsparams)
+
+            # now rotate to direction of (ra_rate, dec_rate)
+            # angle_rad = galsim.Angle(np.arctan2(dec_rate, (ra_rate * np.cos(dec))), galsim.radians)
+            # NOTE: Probably cos(dec) has already been applied, in which
+            # case we want
+            angle_rad = galsim.Angle(np.arctan2(dec_rate, ra_rate),
+                                     galsim.radians)
+
+            gobj = gobj.rotate(angle_rad)
+        else:
+            # Treat as point source
+            return {'this_object': galsim.DeltaFunction(gsparams=gsparams)}
+
```

Comment:
I just noticed that the upper part of the `if/else` doesn't return anything.  The line
```
return {'this_object': gobj}
```
was deleted from the previous version.

Reply:
..which explains an unexpected error I was getting in my test program. Good catch!